From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: George Xu <george.xu@slack-corp.com>
Date: Wed, 11 Dec 2024 15:00:09 -0800
Subject: fix: desktopCapturer.getSources does not return electron window on
 Windows

This patch is to fix issue https://github.com/electron/electron/issues/36037, where `desktopCapturer.getSources` does not return the electron window on Windows.

This patch likely cannot be upstreamed. It can be removed if we can expose the ShouldEnumerateCurrentProcessWindows method in Electron's desktopCapturer API, or if desktopCapturer can use an alternative source API for getSources on Windows.

partially reverts https://chromium-review.googlesource.com/c/chromium/src/+/2907415

diff --git a/content/public/browser/desktop_capture.cc b/content/public/browser/desktop_capture.cc
index 4a806d794aaffe138906f581f72819f19a222e39..a09d1b2115628e29b9c08757ed3cfdf09a6959d2 100644
--- a/content/public/browser/desktop_capture.cc
+++ b/content/public/browser/desktop_capture.cc
@@ -141,11 +141,7 @@ bool CanUsePipeWire() {
 }
 
 bool ShouldEnumerateCurrentProcessWindows() {
-#if BUILDFLAG(IS_WIN)
-  return false;
-#else
   return true;
-#endif
 }
 
 void OpenNativeScreenCapturePicker(
